gravityflow/gravityflowwoocommerce
# Add the Checkout Step

This PR adds a new WooCommerce Checkout Step which can redirect users to the WooCommerce checkout page and the entry will be released from the step after an order created.

We also rename the WooCommerce setting "WooCommerce Integration" to "Create Entry" to better indicate the effect when checking this option. Note that the Checkout Step can only show up in the step types when "Create Entry" isn't checked in the WooCommerce Settings.

New merge tags `{workflow_woocommerce_checkout_link}` and `{workflow_woocommerce_checkout_url}` are also introduced so we can use them in the assignees email of the Checkout step. The merge tag will display a link or URL that leads users to the WooCommerce Checkout page. At the time when users arrive the Checkout page, if there's no product added, they'll see an empty cart and cannot really create an order. To solve this we can use the `gravityflowwoocommerce_checkout_step_url` filter to automatically add a product to cart. See the gist: https://gist.github.com/yoren/4663c9a03a3201c6c20c1deeed38019b for how to use it.

Lastly, because the Payment Step can now be used to pause an "On-hold" order, we must not send assignee emails in such status. This PR changes the logic so when an order was created and paid with a pre-authorized gateway (the order status will be On-hold), we won't send the assignees email but just pause the entry there, until the payment status gets updated to Processing or Completed, we'll then release it.

**Testing instructions**

1. <del>The Checkout step cannot work with our demo site on Pantheon, so you'll need to test it locally</del>. It works now after a native PHP sessions plugin was installed.
2. Import this form for testing: https://www.dropbox.com/s/f8bvg6d3a03ht5e/gravityforms-export-2018-12-20.json?dl=0
3. Update the Checkout Step and set up the "Order Received Redirection" option to your "Workflow Inbox" page, if you'd like to be redirected to the next step after checkout.
4. Use this gist to add a product automatically: https://gist.github.com/yoren/4663c9a03a3201c6c20c1deeed38019b
5. Submit the form, the email you've entered should receive an email with the "Create a WooCommerce Order" link.
6. Click on the link, you should be able to checkout with the specified product and then move onto the User Input step.
